rust/src/passwd.rs

```diff
@@ -475,33 +486,32 @@ fn group_append_unique(
     Ok(())
 }
 
-fn has_usrlib_passwd(rootfs: &openat::Dir) -> Result<bool> {
+fn has_usrlib_passwd(rootfs: &Dir) -> Result<bool> {
     // Does this rootfs have a usr/lib/passwd? We might be doing a
     // container or something else.
-    Ok(rootfs.exists("usr/lib/passwd")?)
+    rootfs.try_exists("usr/lib/passwd").map_err(Into::into)
 }
 
-fn prepare_pwgrp(rootfs: &openat::Dir, merge_passwd_dir: Option<PathBuf>) -> Result<()> {
+fn prepare_pwgrp(rootfs: &Dir, merge_passwd_dir: Option<PathBuf>) -> Result<()> {
     for filename in USRLIB_PWGRP_FILES {
         let etc_file = format!("etc/{}", filename);
         let etc_backup = format!("{}.rpmostreesave", etc_file);
         let usrlib_file = format!("usr/lib/{}", filename);
-        let usrlib_file_tmp = format!("{}.tmp", &usrlib_file);
 
         // Retain the current copies in /etc as backups.
-        rootfs.local_rename(&etc_file, &etc_backup)?;
+        rootfs.rename(&etc_file, rootfs, &etc_backup)?;
 
         // Copy /usr/lib/{passwd,group} -> /etc (breaking hardlinks).
-        rootfs.copy_file(&usrlib_file, &etc_file)?;
+        {
+            let mut src = std::io::BufReader::new(rootfs.open(&usrlib_file)?);
+            rootfs.atomic_replace_with(etc_file, |w| std::io::copy(&mut src, w))?;
+        }
 
-        // Copy the merge's passwd/group to usr/lib (breaking hardlinks).
+        // Copy the merge's passwd/group to usr/lireaking hardlinks).
```

Comment:
```suggestion
        // Copy the merge's passwd/group to usr/lib (breaking hardlinks).
```

Reply:
Ah yeah I think this was editor autosave/cursor movement going awry.